### PR TITLE
KIALI-1140 Add destination services for workload details endpoint

### PR DESCRIPTION
**Pull request by xeviknal**
** Describe the change **
Continuing this reverted PR: https://github.com/kiali/kiali/pull/676

** Issue reference **
https://issues.jboss.org/browse/KIALI-1140

### CODE REVIEW

File: business/layer.go [hunk 1]
```diff
@@ -51,15 +52,16 @@ func NewWithBackends(k8s kubernetes.IstioClientInterface, prom prometheus.Client
 	temporaryLayer.Health = HealthService{prom: prom, k8s: k8s}
 	temporaryLayer.Svc = SvcService{prom: prom, k8s: k8s, businessLayer: temporaryLayer}
 	temporaryLayer.IstioConfig = IstioConfigService{k8s: k8s}
-	temporaryLayer.Workload = WorkloadService{k8s: k8s, prom: prom, ns: NewNamespaceService(k8s)}
 	temporaryLayer.Validations = IstioValidationsService{k8s: k8s, ws: temporaryLayer.Workload}
 	temporaryLayer.App = AppService{k8s: k8s}
 	temporaryLayer.Namespace = NewNamespaceService(k8s)
+	temporaryLayer.Workload = WorkloadService{k8s: k8s, prom: prom, businessLayer: temporaryLayer}
```
Comment (xeviknal) [inline business/layer.go]: Problems started right after changing this line. It makes [business/workloads.go#L215](https://github.com/kiali/kiali/pull/684/commits/dc3cb0a239776ad16154ebba6fef4e9fbded3e28#diff-9b809ab418e2f384025bbe22c950eb75R215) to access to k8s (which it has a nil value).

```log
  | I1128 12:27:24.461122       1 workloads.go:214] k8s var is present? %!(EXTRA bool=false, <nil>)
```

File: business/layer.go [hunk 2]
```diff
@@ -55,7 +55,7 @@ func NewWithBackends(k8s kubernetes.IstioClientInterface, prom prometheus.Client
 	temporaryLayer.Validations = IstioValidationsService{k8s: k8s, ws: temporaryLayer.Workload}
 	temporaryLayer.App = AppService{k8s: k8s}
 	temporaryLayer.Namespace = NewNamespaceService(k8s)
-	temporaryLayer.Workload = WorkloadService{k8s: k8s, prom: prom, businessLayer: temporaryLayer}
+	temporaryLayer.Workload = WorkloadService{k8s: k8s, prom: prom, ns: temporaryLayer.Namespace}
```
Comment (xeviknal) [inline business/layer.go]: It keeps breaking even only adding the specific NamespaceService.

**Comment by hhovsepy** [comment]
Regression check passed.

File: prometheus/client.go [hunk 3]
```diff
@@ -117,6 +125,42 @@ func (in *Client) GetSourceWorkloads(namespace string, namespaceCreationTime tim
...
+func (in *Client) GetDestinationServices(namespace string, namespaceCreationTime time.Time, workloadname string) ([]Service, error) {
+	reporter := "source"
+	if config.Get().IstioNamespace == namespace {
+		reporter = "destination"
+	}
+
+	queryTime := util.Clock.Now()
+	queryInterval := queryTime.Sub(namespaceCreationTime)
+	groupBy := "(destination_service_namespace, destination_service_name, destination_service)"
+	query := fmt.Sprintf("sum(rate(istio_requests_total{reporter=\"%s\",source_workload=\"%s\",source_workload_namespace=\"%s\"}[%vs])) by %s",
```
Comment (lucasponce) [inline prometheus/client.go]: The query used for "GetSourceWorkloads" is
```
delta(istio_requests_total{reporter=\"%s\",destination_service_name=\"%s\",destination_service_namespace=\"%s\"}[%vs])
```
I guess that this one is linked with how the graph is fetching this info, and perhaps in a next step we can change/sync the GetSourceWorkloads (if it makes sense).

**Pull request merged**